### PR TITLE
Begin implementing Redux

### PR DESCRIPTION
**Pull request by Gkjsdll**
- Implement more fetching of data from server

### CODE REVIEW

File: actions/cuisineActions.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  clearSearchLocation: function() {
+    return { type: 'CLEAR_SEARCH_LOCATION' };
+  },
+
+  toggleSearchPrice: function(price) {
+    return { type: 'TOGGLE_SEARCH_PRICE', price }
+  },
+
+  clearSearchPrices: function() {
+    return { type: 'CLEAR_SERACH_PRICES' };
```
Comment (sidrees00) [inline actions/cuisineActions.js:23]: TYPO, fix SERACH to SEARCH

File: actions/cuisineActions.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  },
+
+  toggleSearchRestriction: function(restriction) {
+    return { type: 'TOGGLE_SEARCH_RESTRICTION', restriction }
+  },
+
+  clearSearchRestrictions: function() {
+    return { type: 'CLEAR_SERACH_RESTRICTIONS' };
```
Comment (sidrees00) [inline actions/cuisineActions.js:31]: TYPO, fix SERACH to SEARCH

File: actions/authActions.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+module.exports = {
+  setAuthToken: function(token) {
+    return { type: 'SET_AUTHTOKEN', token };
```
Comment (sidrees00) [inline actions/authActions.js:3]: change 'SET_AUTHTOKEN' to 'SET_AUTH_TOKEN' to maintain consistency

File: actions/authActions.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  },
+
+  setCurrentUser: function(id) {
+    return { type: 'SET_CURRENT_USER', id };
+  },
+
+  clearAuthToken: function() {
+    return { type: 'CLEAR_AUTHTOKEN', token: null };
```
Comment (sidrees00) [inline actions/authActions.js:11]: change 'SET_AUTHTOKEN' to 'SET_AUTH_TOKEN' to maintain consistency

Comment (sidrees00) [inline actions/authActions.js:11]: Omit token: null to maintain consistency

File: actions/authActions.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  },
+
+  clearCurrentUser: function() {
+    return { type: 'CLEAR_CURRENT_USER', id: null };
```
Comment (sidrees00) [inline actions/authActions.js:15]: Omit id: null to maintain consistency

File: reducers/auth/currentUserReducer.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+export default function currentUserReducer(state = null, action) {
+  switch (action.type) {
+    case 'SET_CURRENT_USER':
```
Comment (sidrees00) [inline reducers/auth/currentUserReducer.js]: Is this being set to state?  If so, Object.assign() should be used.

File: reducers/searchReducers.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+      return Object.assign({}, state, { cuisine: action.cuisine });
+      break;
+    case 'SET_SEARCH_LOCATION':
+      return Object.assign({}, state, { location: action.location });
+      break;
+    case 'TOGGLE_SEARCH_RESTRICTION':
+      {
+        let update = {}
+        update[action.restriction] = !state.restrictions[action.restriction];
+        let newRestrictions = Object.assign({}, state.restrictions, update);
```
Comment (sidrees00) [inline reducers/searchReducers.js]: put update in curly braces: {update}

File: screens/ProfileScreen.js [hunk 8]
```diff
@@ -91,9 +92,16 @@ export default class ProfileScreen extends React.Component {
       </ScrollView>
     );
   }
+}
 
+function mapStateToProps(state, ownProps) {
```
Comment (sidrees00) [inline screens/ProfileScreen.js:97]: ownProps can be removed

File: screens/SearchScreen.js [hunk 9]
```diff
@@ -123,43 +181,12 @@ const styles = StyleSheet.create({
...
-  {
-    name: 'Vegetarian',
-    value: 'vegetarian',
-  },
-  {
-    name: 'Vegan',
-    value: 'vegan',
-  },
-];
+function mapStateToProps(state, ownProps) {
```
Comment (sidrees00) [inline screens/SearchScreen.js:186]: ownProps can be removed

**Review by sidrees00 — CHANGES_REQUESTED** [review]
Please address comments.  Whooosh!

File: actions/cuisineActions.js (hunk 1 above)
Comment (Gkjsdll) [reply actions/cuisineActions.js:23]: Woops, thanks!

File: actions/cuisineActions.js (hunk 2 above)
Comment (Gkjsdll) [reply actions/cuisineActions.js:31]: Also woops, also thanks!

File: actions/authActions.js (hunk 3 above)
Comment (Gkjsdll) [reply actions/authActions.js:3]: It never has a hyphen anywhere else, it's normally AuthToken, but the convention here is to use only capital letters and underscores, hence "AUTHTOKEN".

File: actions/authActions.js (hunk 4 above)
Comment (Gkjsdll) [reply actions/authActions.js:11]: It never has a hyphen anywhere else, it's normally AuthToken, but the convention here is to use only capital letters and underscores, hence "AUTHTOKEN". Also, this comment uses an underscore & the other uses a hyphen.

Comment (Gkjsdll) [reply actions/authActions.js:11]: I concur, woops.

File: actions/authActions.js (hunk 5 above)
Comment (Gkjsdll) [reply actions/authActions.js:15]: I concur, woops.

**Pull request merged**